### DISCUSSION TITLE
Oauth with Microsoft does not work

### DISCUSSION THREAD

**Discussion by CogalTek** [pocketbase/pocketbase#3612] [Q&A] [answered]
Hello everybody,

I'm trying to connect via Microsoft, so I saved the information from my Microsoft app and when I click on my connection button it asks me to select a user while showing me the correct information from the app,

The small problem is that I have this error which then appears:

<img width="1250" alt="Capture d’écran 2023-10-26 à 17 11 05" src="https://github.com/pocketbase/pocketbase/assets/91875582/750f6649-b51c-4806-931e-b9246c44cd21">

When installing my app for redirection it was necessary to have an https link so I launched a "ngrok http 8090" which allows me to have a temporary address in https which subsequently redirects requests from Microsoft to pocketbase in http,

<img width="1250" alt="Capture d’écran 2023-10-26 à 17 11 44" src="https://github.com/pocketbase/pocketbase/assets/91875582/f0a47dc2-2b42-40b6-b69a-e22c67bc529b">
<img width="730" alt="Capture d’écran 2023-10-26 à 17 12 12" src="https://github.com/pocketbase/pocketbase/assets/91875582/e9588f2e-4446-431e-ada9-41ba08fdc82e">

So if anyone has an idea on how to solve this problem I would be interested.

**Comment by CogalTek** [comment]
the code : 

```Javascript

<template>
    <div class="page d-flex align-items-center justify-content-center flex-column">
        <div class="card card_connexion d-flex align-items-center justify-content-center">
            <img src="/logo_mercure.png" class="logo" alt="">
            <h5 class="text_bienv" style="text-align: center; margin-bottom: 40%;">Bienvenue dans <br>Mercure</h5>
            <button @click="register" style="padding-left: 100px;padding-right: 100px;" class="btn btn-primary">Connexion</button>
        </div>
        <img src="/gestionbat.jpg" class="gestionbat_logo" alt="gestionbat mercure">
    </div>
</template>

<script setup lang="ts">
    import { ref } from 'vue';
    import PocketBase from 'pocketbase';
    const pb = new PocketBase('http://127.0.0.1:8090/');
    
    const register = async () => {
        const authData = await pb.collection('users').authWithOAuth2({ provider: 'microsoft' });
    };
</script>

```

**Reply by ganigeorgiev** [reply]
Change the url in the SDK initialization to the one provided by ngrok, aka. `new PocketBase("https://c308...")`.

You can also find some links and users provided instructions for the Microsoft OAuth2 setup in https://github.com/pocketbase/pocketbase/discussions/3371.

**Reply by CogalTek** [reply]
ok thank you, I went to look and indeed there are a lot of similarities, I changed my code to put the ngrok address,

The problem is that it still doesn't work, do you have any ideas?

my api receives requests from the web app as we can see here
<img width="730" alt="Capture d’écran 2023-10-26 à 17 35 23" src="https://github.com/pocketbase/pocketbase/assets/91875582/63431ca7-e3b9-4161-b797-dabbde0d859a">

**Comment by CogalTek** [comment]
```Javascript
<template>
    <div class="page d-flex align-items-center justify-content-center flex-column">
        <div class="card card_connexion d-flex align-items-center justify-content-center">
            <img src="/logo_mercure.png" class="logo" alt="">
            <h5 class="text_bienv" style="text-align: center; margin-bottom: 40%;">Bienvenue dans <br>Mercure</h5>
            <button @click="register" style="padding-left: 100px;padding-right: 100px;" class="btn btn-primary">Connexion</button>
        </div>
        <img src="/gestionbat.jpg" class="gestionbat_logo" alt="gestionbat mercure">
    </div>
</template>

<script setup lang="ts">
    import { ref } from 'vue';
    import PocketBase from 'pocketbase';
    const pb = new PocketBase('https://2d6d-46-218-78-30.ngrok.io');
    
    const register = async () => {
        const authData = await pb.collection('users').authWithOAuth2({ provider: 'microsoft' });
    };
</script>
```

<img width="730" alt="Capture d’écran 2023-10-26 à 17 36 02" src="https://github.com/pocketbase/pocketbase/assets/91875582/0098a4c2-802d-4674-9c83-9572f5ba47c6">

<img width="730" alt="Capture d’écran 2023-10-26 à 17 37 03" src="https://github.com/pocketbase/pocketbase/assets/91875582/b2361e4b-22ad-4dba-93f7-b0cb07429f91">

**Reply by CogalTek** [reply]
I don't understand something but I don't know what 😫

**Reply by ganigeorgiev** [reply]
The redirect URI in the above screenshot seems to be different (`https://c308...`) from the one that you use in the SDK initialization (`https://2d6d...`).

Note that "Font-channel logout URL" is something else, and it is not used or required by PocketBase.

**Reply by CogalTek** [reply]
Normally the addresses were updated, it sounds the same and I removed the font-channel logout url,

The problem is that it still doesn't work, I don't see what I did wrong in the configuration, and plus I have a cors error that appeared, is there a way to authorize port 3000 in pocketbase?

**Reply by ganigeorgiev** [reply]
I don't think I'll be able to help you. Double check your OAuth2 app configuration.

You can follow the guides from https://github.com/pocketbase/pocketbase/discussions/3371.

**Reply by CogalTek** [reply]
ok, i will retry, thank you

**Comment by CogalTek** [comment]
Good morning,

I come with news,

I managed to unblock the situation, I started again and installed from scratch, and it worked,

now I have another problem,

here is the error code:

```Javascript

Failed to load resource: the server responded with a status of 400 (Bad Request)
chunk-UJJZ7LWV.js:1436 [Vue warn]: Unhandled error during execution of native event handler 
  at <Index onVnodeUnmounted=fn<onVnodeUnmounted> ref=Ref< Proxy(Object) {__v_skip: true} > > 
  at <RouteProvider key="/" vnode= {__v_isVNode: true, __v_skip: true, type: {…}, props: {…}, key: null, …} route= {fullPath: '/', hash: '', query: {…}, name: 'index', path: '/', …}  ... > 
  at <RouterView name=undefined route=undefined > 
  at <NuxtPage> 
  at <NuxtLayoutProvider layoutProps= {ref: RefImpl} key="default" name="default"  ... > 
  at <NuxtLayout> 
  at <App key=3 > 
  at <NuxtRoot>
warn2 @ chunk-UJJZ7LWV.js:1436
logError @ chunk-UJJZ7LWV.js:1610
handleError @ chunk-UJJZ7LWV.js:1602
(anonyme) @ chunk-UJJZ7LWV.js:1563
Promise.catch (asynchrone)
callWithAsyncErrorHandling @ chunk-UJJZ7LWV.js:1562
invoker @ chunk-UJJZ7LWV.js:9752
RecordService.ts:538 Uncaught (in promise) ClientResponseError 400: Failed to authenticate.
    at RecordService.ts:538:32

```

Is there anything configured?

**Reply by ganigeorgiev** [reply, marked as answer]
The error message unfortunately doesn't really help narrowing the cause of the issue.
You can try/catch the call and print `console.log(err.originalError, err.response)`

You can also start the executable with `--debug` and check if there is a more detailed error message in the stderr output and/or check the failed request log in the _Admin UI > Logs_.

One common reason why OAuth2 authentication can fail if it is the first time signing-up then it will attempt to create a PocketBase user, meaning that the create API rule of the auth collection will be checked.

**Reply by CogalTek** [reply]
Hello again,

I found it was completely stupid, in the configuration of the users table, I had checked "Always require email" and that meant that it didn't work,

so now that it works I posted a video on youtube showing how to do oauth2 with microsoft, this will prevent people from making the same mistakes as me


thanks a lot for your help

**Comment by CogalTek** [comment]
[Oauth2 between pocketbase and microsoft with nuxt3](https://youtu.be/_Jl-vjKt2mQ)
